src/viam/sdk/tests/test_utils.hpp

```diff
@@ -61,11 +61,13 @@ void client_to_mock_pipeline(std::shared_ptr<Resource> mock, F&& test_case) {
     grpc::ChannelArguments args;
     auto test_server = TestServer(server);
     auto grpc_channel = test_server.grpc_in_process_channel(args);
-    ClientType resource_client(mock->name(), grpc_channel);
+
+    auto resource_client = Registry::lookup_resource_client(API::get<ClientType>())
```

Comment:
This is a non-mechanical change, so that tests no longer have a dependency on the concrete `FooClient` type. Instead of directly constructing `ClientType`, we ask the registry to do it for us, then downcast. 

Potentially we should rename the `ClientType` template parameter?

Reply:
Yeah, I think `ResourceType` is probably better at this point.